flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -325,7 +325,7 @@ private void fillMetaDataForBinlogSplit(BinlogSplitMetaEvent metadataEvent) {
                         receivedMetaGroupId,
                         expectedMetaGroupId);
             }
-            requestBinlogSplitMetaIfNeeded(binlogSplit);
+            requestBinlogSplitMetaIfNeeded(uncompletedBinlogSplits.get(binlogSplit.splitId()));
```

Comment:
I don't understand this change, could you explain more ?

Reply:
after fillMetaDataForBinlogSplit, will create  a new binlogSplit in uncompletedBinlogSplits and cover last one , should use  new binlogSplit to  calculate  nextMetaGroupId

Reply:
if use old binlogSplit，it will cause MySqlSourceReader send repeate BinlogSplitMetaRequestEvent and  recevie repeate BinlogSplitMetaEvent

Reply:
@lidoudou1993 Could you add a test for the PR？

Reply:
I recurrence this problem byMySqlSourceExampleTest  in debug model. When set splitMetaGroupSize=5,splitSize=50 and two thousand record in table，it will lead receivedMetaGroupId != expectedMetaGroupId when use old binlogSplit calculate nextMetaGroupId,and MySqlSourceReader need send more BinlogSplitMetaRequestEvent to MySqlSourceEnumerator.
<img width="1642" alt="debug1" src="https://user-images.githubusercontent.com/56642952/187339328-c358cf29-5c64-4e6f-a742-7facc0cd9927.png">

Reply:
if use new binlogSplit to calculate nextMetaGroupId,this problem will not happen
<img width="1601" alt="debug2" src="https://user-images.githubusercontent.com/56642952/187340191-123ea863-b6c9-42b2-b735-1ce0d3f9ade0.png">

Reply:
@leonardBang

Reply:
@lidoudou1993 you're right, I catch your point